lib/metadata/models/object/object.js

```diff
@@ -0,0 +1,465 @@
+        return this;
+    }
+
+    /**
+     * Returns storage class
+     *
+     * @return {string} Storage class
+     */
+    getAmzStorageClass() {
+        return this._data['x-amz-storage-class'];
+    }
+
+    /**
+     * Set server side encryption
+     *
+     * @param {string} serverSideEncryption - Server side encryption
+     * @return {ObjectMD} itself
+     */
+    setAmzServerSideEncryption(serverSideEncryption) {
+        this._data['x-amz-server-side-encryption'] = serverSideEncryption;
+        return this;
+    }
+
+    /**
+     * Returns server side encryption
+     *
+     * @return {string} server side encryption
+     */
+    getAmzServerSideEncryption() {
+        return this._data['x-amz-server-side-encryption'];
+    }
+
+    /**
+     * Set encryption key id
+     *
+     * @param {string} keyId - Encryption key id
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionKeyId(keyId) {
+        this._data['x-amz-server-side-encryption-aws-kms-key-id'] = keyId;
+        return this;
+    }
+
+    /**
+     * Returns encryption key id
+     *
+     * @return {string} Encryption key id
+     */
+    getAmzEncryptionKeyId() {
+        return this._data['x-amz-server-side-encryption-aws-kms-key-id'];
+    }
+
+    /**
+     * Set encryption customer algorithm
+     *
+     * @param {string} algo - Encryption customer algorithm
+     * @return {ObjectMD} itself
+     */
+    setAmzEncryptionCustomerAlgorithm(algo) {
+        this._data['x-amz-server-side-encryption-customer-algorithm'] = algo;
+        return this;
+    }
+
+    /**
+     * Returns Encryption customer algorithm
+     *
+     * @return {string} Encryption customer algorithm
+     */
+    getAmzEncryptionCustomerAlgorithm() {
+        return this._data['x-amz-server-side-encryption-customer-algorithm'];
+    }
+
+    /**
+     * Set access control list
+     *
+     * @param {object} acl - Access control list
+     * @param {string} acl.Canned -
+     * @param {string[]} acl.FULL_CONTROL -
+     * @param {string[]} acl.WRITE_ACP -
+     * @param {string[]} acl.READ -
+     * @param {string[]} acl.READ_ACP -
+     * @return {ObjectMD} itself
+     */
+    setAcl(acl) {
+        this._data.acl = acl;
+        return this;
+    }
+
+    /**
+     * Returns access control list
+     *
+     * @return {object} Access control list
+     */
+    getAcl() {
+        return this._data.acl;
+    }
+
+    /**
+     * Set object key
+     *
+     * @param {string} key - Object key
+     * @return {ObjectMD} itself
+     */
+    setKey(key) {
+        this._data.key = key;
+        return this;
+    }
+
+    /**
+     * Returns object key
+     *
+     * @return {string} object key
+     */
+    getKey() {
+        return this._data.key;
+    }
+
+    /**
+     * Set location
+     *
+     * @param {string[]} location - location
+     * @return {ObjectMD} itself
+     */
+    setLocation(location) {
+        this._data.location = location;
```

Comment:
Keeping this as is due to performance considerations, and since locations should not be modified outside of this class.